File: TradeItIosEmsApi/TradeItResult.m
#import "TradeItResult.h"

@implementation TradeItResult

- (id)init {
    self = [super init];
    if (self) {
        self.token = nil;
        self.shortMessage = nil;
        self.longMessages = nil;
        self.status = nil;
    }
    return self;
}

- (NSString *)description{
    return [NSString stringWithFormat:@"status=%@ token=%@ shortMessage=%@ longMessages%@",self.status, self.token, self.shortMessage, self.longMessages];
}

- (BOOL)isSuccessful {
    return [@"SUCCESS" isEqualToString:self.status] || [@"REVIEW_ORDER" isEqualToString:self.status];
}

- (BOOL)isSecurityQuestion {
    return [@"INFORMATION_NEEDED" isEqualToString:self.status];
}

- (BOOL)isReviewOrder {
    return [@"REVIEW_ORDER" isEqualToString:self.status];
}

- (BOOL)isError {
    return [@"ERROR" isEqualToString:self.status];
}

@end

File: TradeItIosEmsApi/TradeItResult.swift
public class TradeItResult: NSObject, Codable {
    var status: String?
    var token: String?
    public var shortMessage: String?
    public var longMessages: [String]?
    
    func isSuccessful() -> Bool {
        return ["SUCCESS", "REVIEW_ORDER"].contains(self.status)
    }
    
    func isSecurityQuestion() -> Bool {
        return self.status == "INFORMATION_NEEDED"
    }
    
    func isReviewOrder() -> Bool {
        return self.status == "REVIEW_ORDER"
    }
    
    func isError() -> Bool {
        return self.status == "ERROR"
    }
}
